c/src/lib/libbsp/arm/stm32f4/can/can.c

```diff
@@ -0,0 +1,543 @@
+
+  if (HAL_CAN_Init(hCanHandle) != HAL_OK)
+  {
+    /* Initialization Error */
+    //TODO::
+    return CAN_ERROR;
+  }
+  CAN_FilterConfTypeDef sFilterConfig;
+  sFilterConfig.FilterNumber = 0;
+  sFilterConfig.FilterMode = CAN_FILTERMODE_IDMASK;
+  sFilterConfig.FilterScale = CAN_FILTERSCALE_32BIT;
+  sFilterConfig.FilterIdHigh = 0x0000;
+  sFilterConfig.FilterIdLow = 0x0000;
+  sFilterConfig.FilterMaskIdHigh = 0x0000;
+  sFilterConfig.FilterMaskIdLow = 0x0000;
+  sFilterConfig.FilterFIFOAssignment = 0;
+  sFilterConfig.FilterActivation = ENABLE;
+  sFilterConfig.BankNumber = 14;
+
+  if(HAL_CAN_ConfigFilter(&hCanHandle, &sFilterConfig) != HAL_OK) {
+
+  }
+
+  rtems_status_code sc;
+
+  if (CAN_ONE == canInstance)
+  {
+    sc = rtems_interrupt_handler_install(
+    	CAN1_TX_IRQn,
+        "CAN1 TX Interrupt",
+        RTEMS_INTERRUPT_UNIQUE,
+        stm32_can_isr,
+        hCanHandle);
+    sc = rtems_interrupt_handler_install(
+    	CAN1_RX0_IRQn,
+        "CAN1 RX Interrupt 0",
+        RTEMS_INTERRUPT_UNIQUE,
+        stm32_can_rx0_isr,
+        hCanHandle);
+    sc = rtems_interrupt_handler_install(
+    	CAN1_RX0_IRQn,
+        "CAN1 RX Interrupt 1",
+        RTEMS_INTERRUPT_UNIQUE,
+        stm32_can_rx1_isr,
+        hCanHandle);
+  } else 
+  {
+	  sc = rtems_interrupt_handler_install(
+			  CAN2_TX_IRQn,
+			  "CAN2 TX Interrupt",
+			  RTEMS_INTERRUPT_UNIQUE,
+			  stm32_can_isr,
+			  hCanHandle);
+	  sc = rtems_interrupt_handler_install(
+			  CAN2_RX0_IRQn,
+			  "CAN2 RX Interrupt 0",
+			  RTEMS_INTERRUPT_UNIQUE,
+			  stm32_can_rx0_isr,
+			  hCanHandle);
+	  sc = rtems_interrupt_handler_install(
+			  CAN2_RX1_IRQn,
+			  "CAN2 RX Interrupt 1",
+			  RTEMS_INTERRUPT_UNIQUE,
+			  stm32_can_rx1_isr,
+			  hCanHandle);
+      }
+  return CAN_OK;
+}
+
+
+int stm32_can_de_init(
+    can_bus * self
+){
+  stm32_can_bus * bus = (stm32_can_bus * ) self;
+  CAN_HandleTypeDef * hCanHandle = &(bus->wrapper.handle);
+  rtems_status_code sc;
+
+  if (CAN_ONE == hCanHandle->Instance)
+  {
+    sc = rtems_interrupt_handler_remove(
+        19, 
+        stm32_can_isr,     
+        hCanHandle
+        );
+    sc = rtems_interrupt_handler_remove(
+        27, 
+        stm32_can_rx0_isr, 
+        hCanHandle
+        );
+    sc = rtems_interrupt_handler_remove(
+        28, 
+        stm32_can_rx1_isr,
+        hCanHandle);
+  } else 
+  {
+    /* 
+     * TODO figure out CAN 2 interrupts
+    sc = rtems_interrupt_handler_remove(
+        19, 
+        stm32_can_isr,     
+        hCanHandle
+        );
+    sc = rtems_interrupt_handler_install(
+        27, 
+        stm32_can_rx0_isr, 
+        hCanHandle
+        );
+    sc = rtems_interrupt_handler_install(
+        28, 
+        stm32_can_rx1_isr,
+        hCanHandle);
+    */
+  }
+
+  HAL_CAN_DeInit(hCanHandle);
+  // TODO error codes
+  return 0;
+}
+
+stm32_can_bus * bus1;
```

Comment:
Hey Chris, please don't review this file. The files in  c/src/lib/libbsp/arm/stm32f4/ are old BSP that we forgot to remove.